std/array.d

```diff
@@ -2277,48 +2277,47 @@ unittest
     shrinks the array as needed.
  +/
 void replaceInPlace(T, Range)(ref T[] array, size_t from, size_t to, Range stuff)
-    if(isDynamicArray!Range &&
-       is(ElementEncodingType!Range : T) &&
-       !is(T == const T) &&
-       !is(T == immutable T))
+    if(is(typeof(replace(array, from, to, stuff))))
 {
-    import std.algorithm : remove;
-    import std.typecons : tuple;
-
-    if (overlap(array, stuff).length)
+    static if(isDynamicArray!Range &&
+              is(Unqual!(ElementEncodingType!Range) == T) &&
+              !is(T == const T) &&
+              !is(T == immutable T) &&
```

Comment:
I wrote this in unit test, and it didn't fail:

```D
    void foo(inout(int)[] x)
    {
        x.replaceInPlace(1,2, [1,2,3]);
    }
```

Not sure if I did that right, I agree it probably should match the static if.